### ISSUE TITLE
OS_Regex will add extra character when using \.

### ISSUE THREAD

**Issue by rossengeorgiev** [wazuh/wazuh#570]
Here is the a minimal decoder to replicate:
```
<decoder name="regexbug">
  <prematch>^regexbug</prematch>
  <regex>^regexbug \((\.+)\)$</regex>
  <order>name</order>
</decoder>
```

`/var/ossec/bin/ossec-logtest`
```
**Phase 1: Completed pre-decoding.
       full event: 'regexbug (my text here)'
       hostname: 'myhost'
       program_name: '(null)'
       log: 'regexbug (my text here)'

**Phase 2: Completed decoding.
       decoder: 'regexbug'
       name: 'my text here)'
```

**Expected**: the name variable to be `my text here`
**Result**: the name variable includes an extra `)` at the end

**Comment by TJOSERAFAEL** [comment]
Hi @rossengeorgiev ,

as you noticed the regex function is bugged. As a workaround you can use this expression 
`^regexbug \((\.+)\)\)$`.

In Wazuh 4.0 we are going to use a regex library for C that will wipeout all regex bugs that we have now.

Best regards.

**Comment by rossengeorgiev** [comment]
Hi @TJOSERAFAEL, the suggest expression doesn't seem to match at all.

```
**Phase 1: Completed pre-decoding.
       full event: 'regexbug (my text here)'
       hostname: 'myhost'
       program_name: '(null)'
       log: 'regexbug (my text here)'

**Phase 2: Completed decoding.
       decoder: 'regexbug'
```
For clarity, this happens with any character, in my case `"`.

**Comment by rossengeorgiev** [comment]
Was this bug resolved?

**Comment by TJOSERAFAEL** [comment]
Hi @rossengeorgiev ,

we closed this by error. This is a complex issue (we have to replace the entire **regex** engine). When this get fixed we will announce it.

Best regards.

**Comment by jnasselle** [comment]
Hi @rossengeorgiev ,

This was solved since v3.8 by changes introduced in #2224.

Regards,
Nico

**Comment by rossengeorgiev** [comment]
@jnasselle ok, thanks. Closing this
